rest-service/manager_rest/plugin_update/manager.py

```diff
@@ -0,0 +1,265 @@
+
+import os
+import uuid
+
+from flask import current_app
+
+from cloudify.models_states import ExecutionState
+
+from manager_rest import config, utils
+from manager_rest.storage import get_storage_manager, models
+from manager_rest.resource_manager import get_resource_manager
+from manager_rest.plugin_update.constants import (STATES,
+                                                  ACTIVE_STATES)
+from manager_rest.manager_exceptions import (ConflictError,
+                                             PluginsUpdateError,
+                                             IllegalActionError)
+from manager_rest.constants import (FILE_SERVER_BLUEPRINTS_FOLDER,
+                                    HIDDEN_BLUEPRINT_SUFFIX)
+
+from dsl_parser.constants import (PLUGIN_NAME_KEY,
+                                  WORKFLOW_PLUGINS_TO_INSTALL,
+                                  DEPLOYMENT_PLUGINS_TO_INSTALL)
+
+
+class PluginsUpdateManager(object):
+    def __init__(self, sm):
+        self.sm = sm
+        self.rm = get_resource_manager(self.sm)
+
+    def validate_no_active_updates_per_blueprint(self,
+                                                 blueprint_id,
+                                                 force=False):
+        active_updates = self.list_plugins_updates(
+            filters={'blueprint_id': blueprint_id,
+                     'state': ACTIVE_STATES}).items
+        if not active_updates:
+            return
+
+        if not force:
+            raise ConflictError(
+                'There are plugins updates still active, update IDs: '
+                '{0}'.format(', '.join([u.id for u in active_updates])))
+
+        execs_to_updates = {u.execution_id: u for u in active_updates
+                            if u.execution_id}
+        # real active updates are those with an execution in a running status
+        real_active_updates = [
+            execs_to_updates[ex.id]
+            for ex in self.sm.list(models.Execution,
+                                   filters={'id': execs_to_updates.keys()}
+                                   ).items
+            if ex.status not in ExecutionState.END_STATES]
+
+        if real_active_updates:
+            raise ConflictError(
+                'There are plugins updates still active; the "force" flag '
+                'was used yet these updates have actual executions running '
+                'update IDs: {0}'.format(', '.join(
+                    [u.id for u in real_active_updates])))
+        else:
+            # the active updates aren't really active - either their
+            # executions were failed/cancelled, or the update failed at
+            # the finalizing stage.
+            # updating their states to failed and continuing.
+            for plugins_update in active_updates:
+                plugins_update.state = STATES.FAILED
+                self.sm.update(plugins_update)
+
+    @staticmethod
+    def _get_available_random_resource_id(prefix):
```

Comment:
Should be more simple, like prefix+uuid4 (cut this if needed by 256-len pre -len suf)+suffix

Reply:
Resolving after the discussion we had